packages/api/cms-api/src/user-permissions/auth/user-permissions.guard.ts

```diff
@@ -19,38 +19,52 @@ export class UserPermissionsGuard implements CanActivate {
     ) {}
 
     async canActivate(context: ExecutionContext): Promise<boolean> {
-        const location = `${context.getClass().name}::${context.getHandler().name}()`;
+        if (this.getDecorator(context, "disableCometGuards")) {
+            return true;
+        }
 
-        const requiredContentScopes = await this.contentScopeService.getScopesForPermissionCheck(context);
+        const user = this.getUser(context);
 
-        // Ignore field resolvers as they have no scopes and would overwrite the scopes of the root query.
-        if (!this.isResolvingGraphQLField(context)) {
-            const request = getRequestFromExecutionContext(context);
-            request.contentScopes = this.contentScopeService.getUniqueScopes(requiredContentScopes);
+        if (!user) {
+            return false;
         }
 
-        if (this.getDecorator(context, "disableCometGuards")) return true;
+        const location = `${context.getClass().name}::${context.getHandler().name}()`;
 
-        const user = this.getUser(context);
-        if (!user) return false;
+        const requiredPermission = this.getDecorator<RequiredPermissionMetadata>(context, "requiredPermission");
+        const skipScopeCheck = requiredPermission?.options?.skipScopeCheck ?? false;
 
         // System user authenticated via basic auth
-        if (typeof user === "string" && this.options.systemUsers?.includes(user)) return true;
+        if (typeof user === "string" && this.options.systemUsers?.includes(user)) {
+            // Ignore field resolvers as they have no scopes and would overwrite the scopes of the root query.
+            if (!this.isResolvingGraphQLField(context) && !skipScopeCheck) {
+                const contentScopes = await this.contentScopeService.inferScopesFromExecutionContext(context);
```

Comment:
We have a try/catch around this in the [ChangesCheckerInterceptor](https://github.com/vivid-planet/comet/blob/respect-skip-scope-check/packages/api/cms-api/src/builds/changes-checker.interceptor.ts#L33-L39). I'm not sure if we need this here as well, but I'd say we don't. What do you think?